File: servers/PingServer2.ts
// Server configuration
const HOST = "0.0.0.0";
const PORT = 35098  ;

class PingServer {
  private udpServer: Deno.DatagramConn | null = null;

  async start(): Promise<void> {
    try {
      console.log(`[Ping Server] Online! ${HOST}:${PORT}`);
      this.udpServer = await Deno.listenDatagram({ hostname: HOST, port: PORT, transport: "udp" });

      for await (const [data, addr] of this.udpServer) {
        this.datagramReceived(data, addr);
      }
    } catch (error) {
      console.error(`[Ping Server] Error: ${error}`);
    }
  }

  datagramReceived(data: Uint8Array, addr: Deno.Addr): void {
    const message = new TextDecoder().decode(data);
    const netAddr = addr as Deno.NetAddr; // Convert to Deno.NetAddr to access the port
    const port = netAddr.port;
    console.log(`[Ping Server] Received message from ${addr.transport}:${port}: ${message}`);

    // Process the message and perform any game-related logic here
    // Update game state, respond to the client, etc.

    // Send a packet containing the letter "a"
    const responseMessage = new TextEncoder().encode("a");
    this.udpServer?.send(responseMessage, addr);
  }
}

async function main() {
  const server = new PingServer();
  await server.start();
}

if (import.meta.main) {
  main();
}

File: servers/PingServer2.py
from twisted.internet.protocol import DatagramProtocol
from twisted.internet import reactor, task

# Server configuration
HOST = '0.0.0.0'
PORT = 21274

class GameServerProtocol(DatagramProtocol):
    def startProtocol(self):
        print(f'Serving on {HOST}:{PORT}')
        
        # Start sending packets every 1 second
        self.send_task = task.LoopingCall(self.send_packet)
        self.send_task.start(1.0)  # Send a packet every 1 second

    def datagramReceived(self, data, addr):
        print(f"Received message from {addr}: {data}")

        # Process the message and perform any game-related logic here
        # Update game state, respond to the client, etc.

    def send_packet(self):
        # Send a packet containing the letter "a"
        message = b'a'
        self.transport.write(message, ('127.0.0.1', PORT))  # Replace IP with appropriate client IP

def main():
    reactor.listenUDP(PORT, GameServerProtocol(), interface=HOST)
    reactor.run()

if __name__ == '__main__':
    main()